main.py

```diff
@@ -38,12 +38,15 @@ async def fetch(self, url) -> list[dict]:
                     await asyncio.sleep(0.1 * 2**try_no)
                 elif isinstance(e, aiohttp.ClientResponseError):
                     logging.error("%s failed with code %s", url, e.status)
-                    raise e
+                    if e.status == 520:
+                        raise RuntimeError("Cloudflare error message indicating that the origin web server received an invalid or incorrectly interpreted request, resulting in an empty response.")
```

Comment:
Are you sure you don't want to retry on 520?

Reply:
Eh, I would have expected a [`403`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403) for that.

Reply:
Sounds good. Cloudflare's own documentation say it is a problem with the origin server, not a block: https://community.cloudflare.com/t/community-tip-fixing-error-520-web-server-is-returning-an-unknown-error/44205